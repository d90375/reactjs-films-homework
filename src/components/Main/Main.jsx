import React from "react";
import Preloader from "./Preloader";
import Card from "./Card";
import TopSort from "./TopSort";
import "./Main.scss";

const CARD_COUNT = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];

const Main = () => {
  return (
    <div className="main">
      <div className="main__wrap">
        <TopSort />
        <div className="card__wrap">
          {CARD_COUNT.map((card) => (
            <Card key={card.toString()} />
          ))}
        </div>
        <Preloader />
      </div>
    </div>
  );
};

export default Main;
