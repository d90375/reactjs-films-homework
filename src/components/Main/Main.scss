.main__wrap {
  padding-top: 4rem;
  padding-left: 4rem;
  padding-right: 4rem;
}

.main {
  background-color: #f5f6f6;
}
