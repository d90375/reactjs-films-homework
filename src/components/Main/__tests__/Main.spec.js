import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Main from "../Main";

const renderer = new ShallowRenderer();
const setUp = (props) => {
  return renderer.render(<Main {...props} />);
};

describe("Main component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render Main component", () => {
    expect(component).toMatchSnapshot();
  });
});
