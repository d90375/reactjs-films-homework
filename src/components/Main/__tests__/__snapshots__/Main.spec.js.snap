// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render Main component 1`] = `
<section
  className="main"
>
  <div
    className="main__wrap"
  >
    <TopSort />
    <div
      className="card__wrap"
    >
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
    </div>
    <Preloader />
  </div>
</section>
`;
